docs/concepts/input/focus.md

```diff
@@ -0,0 +1,149 @@
+﻿---
+id: focus
+title: Focus
+---
+
+import DirectionalNavigationScreenshot from '/img/concepts/input/directional-navigation.gif';
+
+# Focus
+
+Focus refers to the `InputElement` which is expected to receive keyboard input and are typically 
+distinguished with a visual indicator. The most familiar example is a `TextBox` with a blinking cursor inside, but 
+non-textual controls like `Button` and `Slider` also participate in focus.
+
+## IsFocused and Focusable
+
+`IsFocused` is a get-only property that keeps track of the `InputElement`'s focus state.
+
+The `Focusable` property enables or disables the ability to focus an `InputElement`. Elements which cannot be focused 
+can still be interacted with via pointer, so some care should be taken that a functional keyboard equivalent (such as 
+hotkeys) is available when possible.
+
+## Explicit Focusing
+
+To explicitly assign focus to any `InputElement`, call its `.Focus()` method from code. Optionally, you may specify the 
+`NavigationMethod` and `KeyModifiers` to simulate as if the focus were triggered by a specific focus navigation flow. Explicit 
+focusing is often used to focus a specific `InputElement` in a data entry form upon loading or to programmatically move 
+to the next `InputControl` once the current input has been satisfied.
+
+| NavigationMethod | Trigger Description        |
+|:-----------------|:---------------------------|
+| Tab              | Tab key press              |
+| Pointer          | Pointer interaction        |
+| Directional      | 2D Directional (`XYFocus`) |
+| Unspecified      | Default                    |
+
+## Focus Events
+
+`InputElement`s expose `GotFocus` and `LostFocus` events. The `GotFocusEventArgs` contains the `NavigationMethod` and 
+`KeyModifiers` used to trigger the focus navigation.
+
+## Focus Pseudoclasses
+
+These pseudoclasses are helpful in creating visual indicators when styling `Control`s that are `Focusable`.
+
+| Pseudoclass    | Description                                                                    |
+|:---------------|--------------------------------------------------------------------------------|
+| :focus         | The Control has focus.                                                         |
+| :focus-within  | The Control has focus or contains a descendant that has focus.                 |
+| :focus-visible | The Control has focus and received it from Tab or Directional focus navigation |
```

Comment:
To be more specific, `focus-visible` tells that control should use focus styles. Like, text box changes its appearance when focused. Additionally, if control overrides `:focus-visible`, it might need to set FocusAdorner to null:
https://github.com/AvaloniaUI/Avalonia/blob/0cd220f6c2ae409f5a27c261d09c2eb5836ad2fe/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L107

As otherwise double focus visuals would be applied.